src/tangle.ts

```diff
@@ -124,11 +136,20 @@ export class Client<T> {
             fn(this.defaultValue[eventName]);
         }
 
+        let lastValue: T[K];
         return this.events.pipe(
-            pluck('transient'),
-            pluck(eventName),
-            filter((val) => val !== undefined)
-        ).subscribe(fn);
+            map(event => event?.transient),
+            map(transient => transient?.[eventName] as T[K]),
+            filter(value => typeof value !== 'undefined'),
+            map(value => value as T[K])
+        ).subscribe((val: T[K]) => {
+            if (val === lastValue) {
+                return;
+            }
+
+            lastValue = val;
+            fn(val);
+        });
```

Comment:
You can simplify this with a rxjs-native operator like so:

```suggestion
            map(value => value as T[K])
            distinctUntilChanged()
        ).subscribe(fn);
```

Reply:
This doesn't seem to work. It seems that the initial/first change is not propagated, e.g. this test fails:

```ts
const namespace = 'test1';

    const ch = new Channel<Payload>(namespace, { someProp: 0 } as any);
    const bus = await ch.registerPromise([
        new Worker(workerPath, { argv, workerData: { channel: namespace, add: 1 } })
    ]);
    let result = 0;
    bus.listen('someProp', (sum?: number) => {
        result += sum || 0;
    });

    bus.broadcast({ someProp: 1 });
    bus.broadcast({ someProp: 1 });
    bus.broadcast({ someProp: 1 });
    bus.broadcast({ someProp: 1 });
    bus.broadcast({ someProp: 1 });
    bus.broadcast({ someProp: 1 });
    // implicitly wait to allow events to come in
    await new Promise((resolve) => setTimeout(resolve, 200));

    expect(result).toBe(1);
    ch.providers.map((p) => p.terminate());
```

The state change listener is not called once even though it should at the first time where state changes from `0` to `1`. This causes this and other test to fail and is not the behavior I would expect.

Reply:
Can you commit/push the changes you made to the branch please? The ones that make the test case fail. I can't reproduce the behavior with `distinctUntilChanged()`. Figured it'd be easier to troubleshoot off the same branch.

Reply:
@christian-bromann maybe i'm overlooking something